src/bus/bus.c

```diff
@@ -81,14 +81,25 @@ Peer *bus_find_peer_by_name(Bus *bus, Name **namep, const char *name_str) {
         return peer;
 }
 
-static int bus_log_commit_policy(Bus *bus, const char *action, uint64_t sender_id, uint64_t receiver_id, NameSet *sender_names, NameSet *receiver_names, Message *message) {
+static int bus_log_commit_policy(Bus *bus, const char *action, const char *policy_type, const char *sender_label, const char *receiver_label, uint64_t sender_id, uint64_t receiver_id, NameSet *sender_names, NameSet *receiver_names, Message *message) {
         Log *log = bus->log;
         int r;
 
         message_log_append(message, log);
 
         log_appendf(log, "DBUS_BROKER_TRANSMIT_ACTION=%s\n", action);
 
+        if (policy_type)
+                log_appendf(log, "DBUS_BROKER_POLICY_TYPE=%s\n", policy_type);
+
+        if (sender_label)
+                log_appendf(log, "DBUS_BROKER_SENDER_SECURITY_LABEL=%s\n",
+                            sender_label);
+
+        if (receiver_label)
+                log_appendf(log, "DBUS_BROKER_RECEIVER_SECURITY_LABEL=%s\n",
+                            receiver_label);
```

Comment:
So we finally treat security labels as strings? I am not sure what our last agreement there was. We always tried avoiding this, but it might just not be worth it. Anyway, just to be clear: From now on we follow what selinux does and treat the security-label-output as a zero-terminated string (without any internal zeros)?

Reply:
As far as I'm aware we are guaranteed no internal zeros. Otherwise it could be garbage, but just for logging I think that's something we can deal with if it happens